src/features/recovery/components/ExecuteRecoveryButton/index.tsx

```diff
@@ -22,37 +17,17 @@ export function ExecuteRecoveryButton({
   recovery: RecoveryQueueItem
   compact?: boolean
 }): ReactElement {
-  const { setSubmitError } = useContext(RecoveryListItemContext)
   const { isExecutable, isNext, isPending } = useRecoveryTxState(recovery)
-  const onboard = useOnboard()
-  const wallet = useWallet()
-  const { safe } = useSafeInfo()
   const isDisabled = !isExecutable || isPending
   const isWrongChain = useIsWrongChain()
   const chain = useCurrentChain()
+  const { setTxFlow } = useContext(TxModalContext)
 
   const onClick = async (e: SyntheticEvent) => {
     e.stopPropagation()
     e.preventDefault()
 
-    if (!onboard || !wallet) {
-      return
-    }
-
-    try {
-      await dispatchRecoveryExecution({
-        provider: wallet.provider,
-        chainId: safe.chainId,
-        args: recovery.args,
-        delayModifierAddress: recovery.address,
-        signerAddress: wallet.address,
-      })
-    } catch (_err) {
-      const err = asError(_err)
-
-      trackError(Errors._812, e)
-      setSubmitError(err)
-    }
+    setTxFlow(<RecoveryAttemptFlow params={recovery} />)
```

Comment:
Its not part of this PR, but just to bring one idea if it wasn't noticed yet:

It would be nice if we could change the way how we set the current transaction flow component, like setting a enum on the state or any other primitive value instead of the component call would save us some undesired reconciliations and re-renders because react doesn't memoize non-primitive states.